tools/generators/migrate-converged-pkg/index.ts

```diff
@@ -177,21 +193,71 @@ function normalizeOptions(host: Tree, options: MigrateConvergedPkgGeneratorSchem
   };
 }
 
-function validateUserInput(tree: Tree, options: MigrateConvergedPkgGeneratorSchema) {
+function validateUserInput(tree: Tree, options: MigrateConvergedPkgGeneratorSchema): asserts options is AssertedSchema {
   if (!options.name) {
     throw new Error(`--name cannot be empty. Please provide name of the package.`);
   }
 
   const projectConfig = readProjectConfiguration(tree, options.name);
-  const packageJson = readJson<PackageJson>(tree, joinPathFragments(projectConfig.root, 'package.json'));
-
-  const isPackageConverged = packageJson.version.startsWith('9.');
 
-  if (!isPackageConverged) {
+  if (!isPackageConverged(tree, projectConfig)) {
     throw new Error(
       `${options.name} is not converged package. Make sure to run the migration on packages with version 9.x.x`,
     );
   }
+}
+
+function printStats(tree: Tree, options: MigrateConvergedPkgGeneratorSchema) {
+  const allProjects = getProjects(tree);
+  const stats = {
+    migrated: [] as Array<ProjectConfiguration & { projectName: string }>,
+    notMigrated: [] as Array<ProjectConfiguration & { projectName: string }>,
+  };
+
+  allProjects.forEach((project, projectName) => {
+    if (!isPackageConverged(tree, project)) {
+      return;
+    }
+
+    if (isProjectMigrated(project)) {
+      stats.migrated.push({ projectName, ...project });
+
+      return;
+    }
+    stats.notMigrated.push({ projectName, ...project });
+  });
+
+  logger.info('Convergence DX migration stats:');
+  logger.info('='.repeat(80));
+
+  logger.info(`Migrated (${stats.migrated.length}):`);
+  logger.info(stats.migrated.map(projectStat => `- ${projectStat.projectName}`).join('\n'));
+
+  logger.info('='.repeat(80));
+  logger.info(`Not migrated (${stats.notMigrated.length}):`);
+  logger.info(stats.notMigrated.map(projectStat => `- ${projectStat.projectName}`).join('\n'));
+
+  return tree;
+}
+
+function isPackageConverged(tree: Tree, project: ProjectConfiguration) {
+  const packageJson = readJson<PackageJson>(tree, joinPathFragments(project.root, 'package.json'));
+  return packageJson.version.startsWith('9.');
+}
+
+function isProjectMigrated<T extends ProjectConfiguration>(
+  project: T,
+): project is T & Required<Pick<ProjectConfiguration, 'tags' | 'sourceRoot'>> {
+  // eslint-disable-next-line eqeqeq
+  return project.sourceRoot != null && Boolean(project.tags?.includes('vNext'));
```

Comment:
will this tag still be useful after all packages are migrated ? then I imagine we'll go back to a v9 validation

Reply:
they will, we can enforce lint rules based on those - restrict importing anything to vNext  besides vNext for example

Reply:
to provide more context: nx.json and workspace.json gonna be main source of truth for any kind of project config metadata. (while we are in hybrid mode for convergence and non convergence, this approach is the best I could think of for now)